### ISSUE TITLE
[clang-format] Wrong indentation when `if` is without brackets

### ISSUE THREAD

**Issue by jajik** [llvm/llvm-project#98012]
Maybe I'm missing some configuration option, but given the following code from [openssl](https://github.com/openssl/openssl/blob/82a13a1f5053462f826bfb90061f0f77e3cc98a5/crypto/context.c#L198)

```c
#ifndef OPENSSL_NO_THREAD_POOL
    ctx->threads = ossl_threads_ctx_new(ctx);
    if (ctx->threads == NULL)
        goto err;
#endif
   
        /* Low priority. */
#ifndef FIPS_MODULE
    ctx->child_provider = ossl_child_prov_ctx_new(ctx);
    if (ctx->child_provider == NULL)
        goto err;
#endif
```

the comment gets indented with the `goto` and not with `ctx->child_provider = ...` as I would expect.

**Comment by jajik** [comment]
Configuration used

```
Language: Cpp
IndentWidth: 4
PointerAlignment: Right
ColumnLimit: 80
AlignConsecutiveMacros: AcrossEmptyLinesAndComments
AllowAllArgumentsOnNextLine: true
AllowShortIfStatementsOnASingleLine: Never
AllowShortLoopsOnASingleLine: false
BreakBeforeBraces: Custom
BraceWrapping:
  AfterFunction: true
  AfterStruct: false
  AfterUnion: true
InsertBraces: false
IndentCaseLabels: false
IndentGotoLabels: true
IndentExternBlock: false
MaxEmptyLinesToKeep: 2
QualifierAlignment: Left
SpaceAfterCStyleCast: false
SpaceAfterLogicalNot: false
SpaceBeforeParens: Custom
SpaceBeforeParensOptions:
  AfterControlStatements: true
  AfterFunctionDeclarationName: false
  AfterFunctionDefinitionName: false
  BeforeNonEmptyParentheses: false
UseTab: Never
IncludeBlocks: Preserve
SortIncludes: Never
ReflowComments: false
TypenameMacros: ['STACK_OF']
```

**Comment by rymiel** [comment]
This has already been fixed by cddb9ce04e68d876de895d39d60b6af34d569012. The fix will be present in clang-format 19. Duplicate of #45002.

**Comment by jajik** [comment]
Sorry for the noise, I overlooked it.
